### DISCUSSION TITLE
Can't save as stl file

### DISCUSSION THREAD

**Discussion by RBehera87** [pyvista/pyvista#1471] [Q&A] [answered]
Hi,
I'm creating multiple spheres with different radius using the code below 
```py
import vtk
import pyvista as pv
import numpy as np


with open('particle.1000.dump', 'r') as f:
    line = f.readlines()[9:]
    x1 = [float(line.split()[1]) for line in line]
    y1 = [float(line.split()[2]) for line in line]
    z1 = [float(line.split()[3]) for line in line]
    m1 = [float(line.split()[4]) for line in line]


x = np.array(x1)
y = np.array(y1)
z = np.array(z1)
r = np.array(m1)

color = 'yellow'
p = pv.Plotter()
p.background_color = 'white'

for i, j, m, s in zip(x,y,z,r):
    center = (i, j, m)
    radius = s
    mesh = pv.Sphere(center = center, radius = s)
    p.add_mesh(mesh, color=color, opacity=0.5, show_edges=True) 
    mesh.save('mesh.stl', binary=False)
p.show()
```

I can visualize my geometry but the STL file does not save all of my spheres, but only one.  Any help?

**Comment by RBehera87** [comment]
![Screenshot from 2021-06-30 11-59-59](https://user-images.githubusercontent.com/86717147/123943216-37d89b00-d99c-11eb-91df-d2b1ee5dd759.png)

**Reply by RBehera87** [reply]
Here is my geometry.

**Comment by MatthewFlamm** [comment, marked as answer]
Can you format your code correctly using triple backticks?  I suspect that you are overwriting your stl file inside your for loop or it is outside the for loop and it is only written once.  Try:

```python
for i_sphere, (i, j, m, s) in enumerate(zip(x,y,z,r)):
    center = (i, j, m)
    radius = s
    mesh = pv.Sphere(center = center, radius = s)
    p.add_mesh(mesh, color=color, opacity=0.5, show_edges=True)
    mesh.save(f'mesh{i_sphere}.stl', binary=False)
p.show()
```

Also you could consider making a single multiblock mesh using [pyvista.MultiBlock](https://docs.pyvista.org/core/composite.html#pyvista.MultiBlock) and then saving it as a single `.vtm` file.

**Reply by RBehera87** [reply]
Thank you. It helps. Can I save the final multiblock mesh as STL?

**Reply by banesullivan** [reply]
> Can I save the final multiblock mesh as STL?

Not without merging the multiblock first. See https://docs.pyvista.org/api/core/_autosummary/pyvista.MultiBlock.combine.html#pyvista.MultiBlock.combine and then you'd have to use an `extract_surface` filter
